2a_model/src/models/Snakefile_base.smk

```diff
@@ -27,6 +27,46 @@ rule as_run_config:
         asRunConfig(config,output[0])
 
 
+rule prep_io_data:
+    input:
+        "../../../out/well_obs_inputs.zarr",
+        "../../../out/well_obs_targets.zarr",
+    output:
+        "{outdir}/prepped.npz"
+    run:
+        prep_all_data(x_data_file=input[0],
+                      y_data_file=input[1],
+                      x_vars=config['x_vars'],
+                      y_vars_finetune=config['y_vars'],
+                      spatial_idx_name='site_id',
+                      time_idx_name='date',
+                      train_start_date=config['train_start_date'],
+                      train_end_date=config['train_end_date'],
+                      val_start_date=config['val_start_date'],
+                      val_end_date=config['val_end_date'],
+                      test_start_date=config['test_start_date'],
+                      test_end_date=config['test_end_date'],
+                      val_sites=config['validation_sites'],
```

Comment:
This line was not here before. So even though we specified the "validation_sites" in the configuration file, those sites were not being passed to the `prep_all_data` function which means they were not removed from the training data.